components/addFileButton/index.jsx

```diff
@@ -216,23 +219,36 @@ export default function AddFileButton({ className, onSave, resourceList }) {
   }, [confirmationSetup, confirmed, onConfirmation, file, open]);
 
   return (
-    // eslint-disable-next-line jsx-a11y/label-has-associated-control
-    <label
-      data-testid={TESTCAFE_ID_UPLOAD_BUTTON}
-      disabled={isUploading}
+    <button
```

Comment:
The combination of button > label > input is bad. Wouldn't adding `tabIndex={0}` to the label be a better option, and add a onKeyUp event handler that triggers the input field when Enter is clicked? E.g.

```
const inputRef = createRef();
```

In combination with:

```
    <label
      data-testid={TESTCAFE_ID_UPLOAD_BUTTON}
      disabled={isUploading}
      className={className}
      tabIndex={0}
      onKeyUp={(event) => {
        if (event.code !== "Enter") return;
        inputRef.current.click();
      }}
      role="button"
    >
      {isUploading ? "Uploading..." : "Upload File"}
      <input
        data-testid={TESTCAFE_ID_UPLOAD_INPUT}
        type="file"
        style={{ display: "none" }}
        onClick={(e) => {
          e.target.value = null;
        }}
        onChange={onFileSelect}
        ref={inputRef}
      />
    </label>
```

(I've tried this out, and ESLint returns an error "jsx-a11y/no-noninteractive-element-to-interactive-role" for `role="button"`; I think it's better to just ignore that one for this case.)

This should also allow you to test out the event handler doing `userEvent.type(getByTestId(TESTCAFE_ID_UPLOAD_BUTTON), "{enter}")`, perhaps with an event listener added to the click event of the input field itself to see that it has been triggered.

Reply:
@megoth changed it back to label and ignored two warnings for interactive elements (tabIndex and interactions).